.github/workflows/ci_writer.yml

```diff
@@ -0,0 +1,161 @@
+                },
+                'arm': {
+                  'old': parseFloat(${{ steps.old-bins.outputs.LINUX_ARM_KIB }}),
+                  'cur': parseFloat(${{ steps.new-bins.outputs.LINUX_AMD_KIB }}),
+                },
+              },
+              'windows': {
+                'amd': {
+                  'old': parseFloat(${{ steps.old-bins.outputs.WINDOWS_AMD_KIB }}),
+                  'cur': parseFloat(${{ steps.new-bins.outputs.WINDOWS_AMD_KIB }}),
+                },
+              },
+            };
+
+            const lastRelease = '${{ steps.latestrelease.outputs.VERSION_TAG }}';
+            let msg = `
+              🍕 Here are the new binary sizes!
+
+              | Name | New size (kiB) | ${lastRelease} size (kiB) | Delta (%) |
+              |:--   |:---            |:---                    |:--    |
+            `;
+
+            let shouldAcknowledgeSize = false;
+            for (const os of Object.keys(dat)) {
+              for (const arch of Object.keys(dat[os])) {
+                const delta = diff(dat[os][arch].old, dat[os][arch].cur);
+                if (delta > 10) {
+                  shouldAcknowledgeSize = true;
+                }
+            
+                let symbol = "";
+                if (delta > 5) {
+                  symbol = "😭 +";
+                } else if (delta > 1) {
+                  symbol = "🥺 +"
+                } else if (delta > 0) {
+                  symbol = "+";
+                } else if (delta < 0) {
+                  symbol = "❤️ ";
+                }
+                msg += `| ${os} (${arch}) | ${dat[os][arch].cur} | ${dat[os][arch].old} | ${symbol}${delta} |` + "\n";
+              }
+            }
+            
+            if (shouldAcknowledgeSize) {
+              msg += "**The binary size increased more than expected! Applying the `do-not-merge` label.**\n"
+              github.rest.issues.addLabels({
+                issue_number: context.issue.number,
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                labels: ['do-not-merge']
+              });
+            }
+
+            return msg;
+
+      - name: Find comment
+        uses: peter-evans/find-comment@v2
+        id: fc
+        with:
+          issue-number: ${{ github.event.pull_request.number }}
+          comment-author: 'github-actions[bot]'
+          body-includes: 'Here are the new binary sizes'
+
+      - name: Create comment if it does not exist
+        if: steps.fc.outputs.comment-id == ''
+        uses: peter-evans/create-or-update-comment@v2
+        with:
+          issue-number: ${{ github.event.pull_request.number }}
+          body: ${{steps.diff.outputs.result}}
+
+      - name: Update comment if it exists
+        if: steps.fc.outputs.comment-id != ''
+        uses: peter-evans/create-or-update-comment@v2
+        with:
+          comment-id: ${{ steps.fc.outputs.comment-id }}
+          body: ${{steps.diff.outputs.result}}
```

Comment:
```suggestion
      - name: Create or update comment
        uses: peter-evans/create-or-update-comment@v2
        with:
          comment-id: ${{ steps.fc.outputs.comment-id }}
          issue-number: ${{ github.event.pull_request.number }}
          body: ${{steps.diff.outputs.result}}
```

Reply:
oh also maybe need to set
`edit-mode: replace`

Reply:
Thanks for the great feedback, I addressed them all!